modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -611,6 +613,7 @@ public class IgniteImpl implements Ignite {
 
         LongSupplier partitionIdleSafeTimePropagationPeriodMsSupplier = partitionIdleSafeTimePropagationPeriodMsSupplier(replicationConfig);
 
+        Marshaller raftMarshaller = new ThreadLocalPartitionCommandsMarshaller(clusterSvc.serializationRegistry());
```

Comment:
I think it will be better to encapsulate this logic in ReplicaManager itself (as it was earlier in TableManager)

Reply:
Create a ticket for this activity https://issues.apache.org/jira/browse/IGNITE-22222